### PR TITLE
[6.1] Vector ref assembly fixes and enable vector and json tests

### PR DESCRIPTION
**Pull request by apoorvdeshmukh**
## Description

Ports #3521 and enable vector and json test coverage through port of #3559.

## Testing
Adds a testcase for validation of ref assembly APIs for SqlVector.

## Guidelines

Please review the contribution guidelines before submitting a pull request:

- [Contributing](/CONTRIBUTING.md)
- [Code of Conduct](/CODE_OF_CONDUCT.md)
- [Best Practices](/policy/coding-best-practices.md)
- [Coding Style](/policy/coding-style.md)
- [Review Process](/policy/review-process.md)

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/VectorTest/VectorAPIValidationTest.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            // Validate ctor2 with ReadOnlyMemory<T> : public SqlVector(ReadOnlyMemory<T> memory) { }
+            vector = new SqlVector<float>(new ReadOnlyMemory<float>(VectorFloat32TestData.testData));
+            Assert.Equal(VectorFloat32TestData.testData, vector.Memory.ToArray());
+            Assert.Equal(3, vector.Length);
+
+            //Validate IsNull property
+            Assert.False(vector.IsNull, "IsNull should be false for non-null vector.");
+
+            // Validate Null property returns null
+            Assert.Null(SqlVector<float>.Null);
```
[the remark this answers is not in the document]
Comment (apoorvdeshmukh) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/VectorTest/VectorAPIValidationTest.cs:36]: `SQlVector<T>.Null` returns `null` hence the assertion.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/VectorTest/VectorAPIValidationTest.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        // Unit tests are covered under SqlVectorTest.cs
+        [Fact]
+        public void VectorAPITest()
+        {
+            // Validate that SqlVector<float> is a valid type and has valid SqlDbType
+            Assert.True(typeof(SqlVector<float>).IsValueType, "SqlVector<float> should be a value type.");
+            Assert.Equal(36, (int)SqlDbTypeExtensions.Vector);
+
+            // Validate ctor1 with float[] : public SqlVector(System.ReadOnlyMemory<T> memory) { }
+            var vector = new SqlVector<float>(VectorFloat32TestData.testData);
```
Comment (mdaigle) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/VectorTest/VectorAPIValidationTest.cs:23]: Nit: Sharing test data across classes like this can make it hard to update tests in the future. It's fine to replicate simple test data. This method could also be broken out into individual tests for the float[] constructor, ROM constructor, null vector, and create null method.

**Review by mdaigle — APPROVED** [review]

**Review by paulmedynski — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/VectorTest/VectorAPIValidationTest.cs (hunk 2 above)
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/VectorTest/VectorAPIValidationTest.cs:23]: @apoorvdeshmukh please address this in 'main'. I will proceed with this PR in release/6.1 branch.
